platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/describer/RequestedConfigurationNotFoundFailureDescriber.java

```diff
@@ -35,6 +36,16 @@ public ConfigurationSelectionException describeFailure(RequestedConfigurationNot
     }
 
     private String buildConfigurationNotFoundFailureMsg(RequestedConfigurationNotFoundFailure failure) {
-        return String.format("A dependency was declared on configuration '%s' which is not declared in the descriptor for %s.", failure.getRequestedName(), failure.getRequestedComponentDisplayName());
+        String additionalMessage = "";
+        boolean isLocalComponent = failure.getRequestedComponentId() instanceof ProjectComponentIdentifier;
```

Comment:
🤔 I know we discussed changing this around a little to remove the `instanceof` check.  Just leave it for now, I'm working on the failure hierarchy and will adjust it then.